From d6973fe8e09cfc988e0b057910c79f2ca3a11d95 Mon Sep 17 00:00:00 2001
From: Christophe Priouzeau <christophe.priouzeau@foss.st.com>
Date: Fri, 1 Sep 2023 11:08:59 +0200
Subject: [PATCH 2/4] GCC: core_mmu_get_mem_by_type -Werror=enum-int-mismatch

| core/mm/core_mmu.c:1514:6: error: conflicting types for 'core_mmu_get_mem_by_type' due to enum/integer mismatch; have 'void(unsigned int,  vaddr_t *, vaddr_t *)' {aka 'void(unsigned int,  long unsigned int *, long unsigned int *)'} [-Werror=enum-int-mismatch]
|  1514 | void core_mmu_get_mem_by_type(unsigned int type, vaddr_t *s, vaddr_t *e)
|       |      ^~~~~~~~~~~~~~~~~~~~~~~~
| In file included from core/include/mm/tee_pager.h:13,
|                  from core/include/mm/fobj.h:11,
|                  from core/include/mm/sp_mem.h:10,
|                  from core/arch/arm/include/kernel/secure_partition.h:14,
|                  from core/include/kernel/user_mode_ctx.h:11,
|                  from core/mm/core_mmu.c:19:
| core/include/mm/core_mmu.h:535:6: note: previous declaration of 'core_mmu_get_mem_by_type' with type 'void(enum teecore_memtypes,  vaddr_t *, vaddr_t *)' {aka 'void(enum teecore_memtypes,  long unsigned int *, long unsigned int *)'}
|   535 | void core_mmu_get_mem_by_type(enum teecore_memtypes type, vaddr_t *s,
|       |      ^~~~~~~~~~~~~~~~~~~~~~~~

Signed-off-by: Christophe Priouzeau <christophe.priouzeau@foss.st.com>
---
 core/mm/core_mmu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/mm/core_mmu.c b/core/mm/core_mmu.c
index 3829f3005..e38c155b2 100644
--- a/core/mm/core_mmu.c
+++ b/core/mm/core_mmu.c
@@ -1511,7 +1511,7 @@ static void *map_pa2va(struct tee_mmap_region *map, paddr_t pa, size_t len)
 /*
  * teecore gets some memory area definitions
  */
-void core_mmu_get_mem_by_type(unsigned int type, vaddr_t *s, vaddr_t *e)
+void core_mmu_get_mem_by_type(enum teecore_memtypes type, vaddr_t *s, vaddr_t *e)
 {
 	struct tee_mmap_region *map = find_map_by_type(type);
 
-- 
2.34.1

